activate.c

```diff
@@ -76,7 +76,7 @@ static void activate_mapping(struct irq_info *info, void *data __attribute__((un
 
 	cpumask_scnprintf(buf, PATH_MAX, applied_mask);
 	ret = fprintf(file, "%s", buf);
-	if (ret < 0) {
+	if (ret < 0 || fflush(file)) {
```

Comment:
Hi there,

would it be possible to log the value of `strerror(errno)` in the warning message as well?

Reply:
Also, it may be better to check for `fflush(file) < 0`.

Reply:
Hi Robin, thanks for your comments! Since Neil has merged the patch, for 1) I think we can get it improved later if users really care about the failing reason of IRQs reassigning. for 2) according to man 3 fflush, when successful 0 is returned, non-zero is other cases. So zero checking and negative checking are same to me. Thanks!

Reply:
I think it is crucial to know what is the **exact** reason why the IRQ affinity cannot be enforced.

About `fflush(3)`, this is not exactly what I have in my man page (fedora 38):

```
RETURN VALUE
       Upon successful completion 0 is returned.
       Otherwise, EOF is returned and errno is set to indicate the error.
```

By the way, I wonder if there could also be an error returned by `fclose` which is currently ignored:

https://github.com/Irqbalance/irqbalance/blob/v1.9.2/activate.c#L84

Would that be OK if I send another PR with these two minor changes?

Reply:
Here it is #266